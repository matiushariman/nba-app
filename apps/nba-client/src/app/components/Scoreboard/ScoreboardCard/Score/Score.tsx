import { ScoreProps } from './Score.types';

const Score = ({ score }: ScoreProps) => (
  <div className="w-[52px]">
    <p className="font-bold text-3xl">{score}</p>
  </div>
);

export default Score;
