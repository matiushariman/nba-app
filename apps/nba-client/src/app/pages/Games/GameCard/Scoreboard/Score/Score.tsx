import { useGameStatusContext } from '../../../../../context/GameStatusContext';

import type { ScoreProps } from './Score.types';

const Score = ({ score }: ScoreProps) => {
  const { hasGameStarted } = useGameStatusContext();

  return (
    <div className="w-[52px]">
      <p className="font-bold text-3xl text-center">
        {hasGameStarted ? score : '-'}
      </p>
    </div>
  );
};

export default Score;
