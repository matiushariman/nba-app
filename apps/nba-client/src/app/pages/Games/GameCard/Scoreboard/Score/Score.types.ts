import type { Team } from '@nba-app/types-nba';

export type ScoreProps = Pick<Team, 'score'>;
