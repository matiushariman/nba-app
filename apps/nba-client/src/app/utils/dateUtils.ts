import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
