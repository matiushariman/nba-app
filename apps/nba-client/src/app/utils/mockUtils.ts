export const shouldUseMock = process.env['NODE_ENV'] === 'development';
