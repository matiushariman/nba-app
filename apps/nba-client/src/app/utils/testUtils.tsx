import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
