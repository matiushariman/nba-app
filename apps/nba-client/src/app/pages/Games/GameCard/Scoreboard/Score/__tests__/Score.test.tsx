import Score from '..';

import { render } from '../../../../../utils/testUtils';

describe('pages/games/GameCard/Scoreboard/GameCard/Score', () => {
  it('should render Score that matches snapshot', () => {
    const { baseElement } = render(<Score score={100} />);

    expect(baseElement).toMatchSnapshot();
  });
});
