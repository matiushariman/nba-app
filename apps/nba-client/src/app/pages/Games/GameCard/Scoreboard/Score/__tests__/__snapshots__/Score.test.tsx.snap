// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/games/GameCard/Scoreboard/GameCard/Score should render Score that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="w-[52px]"
    >
      <p
        class="font-bold text-3xl text-center"
      >
        100
      </p>
    </div>
  </div>
</body>
`;
